sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AzureContainerRegistryClientExtended.cs

```diff
@@ -57,24 +57,26 @@ internal AzureContainerRegistryClient(System.Uri loginUri, ServiceClientCredenti
         /// </exception>
         public AzureContainerRegistryClient(string loginUrl, ServiceClientCredentials credentials, params DelegatingHandler[] handlers) : this(handlers)
         {
-            //Removes issues with clients potentially setting an incorrect / at the end
+            if (credentials == null)
+            {
+                throw new System.ArgumentNullException("credentials");
+            }
+
+            // prevents issues if caller sets an incorrect '/' at the end
             if (loginUrl.EndsWith("/")) {
                 loginUrl = loginUrl.Substring(0, loginUrl.Length - 1);
             }
 
-            //Removes issues with clients forgetting to set the http entry point
+            // prevents issues if caller does not prefix url with "http" 
             if (!loginUrl.ToLower().StartsWith("http"))
             {
-                loginUrl = "https://" + loginUrl;
+                loginUrl = $"https://{loginUrl}";
             }
 
+            // set fields / properties
             LoginUri = loginUrl;
-
-            Credentials = credentials ?? throw new System.ArgumentNullException("credentials");
-            if (Credentials != null)
-            {
-                Credentials.InitializeServiceClient(this);
-            }
+            Credentials = credentials; 
+            Credentials.InitializeServiceClient(this);
```

Comment:
@estebanreyl  This is what I was talking about, the client's constructor calls this method. Which is defined above in ACRClientCredentials.